Rozdział 10/klikaczpro.html

```diff
@@ -0,0 +1,77 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>Document</title>
+</head>
+<body>
+    <h1 id="heading">Hello world!</h1>
+    
+    <script src="https://code.jquery.com/jquery-3.3.1.min.js"></script>
+    
+    <script>
+    
+   var offset = 0;
+   var direction ='right';
+   var clicks = 50;
+   var counter = 0;
+   
+   $("#heading").offset({left: offset, top: offset});
+
+   var movingHeading = function () {
+    if (direction === 'right') {
+        $("#heading").offset({left: offset});
+        offset++;
+        if (offset > 200) {
+            offset = 0;
+            direction = 'down';
+        } 
+    } else if (direction === 'down') {
+        $("#heading").offset({top: offset});
+        offset++;
+        if (offset > 200) {
+            offset = 200;
+            direction = 'left';
+        }
+    } else if (direction === 'left') {
+        $("#heading").offset({left: offset});
+        offset--;
+        if (offset < 0) {
+            offset = 200;
+            direction = 'up';
+        }
+    } else if (direction === 'up') {
+        $("#heading").offset({top: offset});
+        offset--;
+        if (offset < 0) {
+            offset = 0;
+            direction = 'right';
+        }
+    }
+} 
+
+
+var intervalId = setInterval(movingHeading, clicks);
+
+$("#heading").click(function (event) {
```

Comment:
cała ta część działa trochę przez przypadek, zobacz `clicks /= 1,5;` w js nie zapisujesz sobie numerków z użyciem przecinków, tylko kropek. To co tu się dzieje to dzielenie clicks przez 1 to `,5` jest całkowicie ignorowane.

Dlaczego w takim razie przyspiesza nam ten heading? a no dlatego, ze nie czyścimy dobrze intervala.

Linijka 62: `var intervalId = setInterval(movingHeading, clicks);` tworzy nowy interval. Nie chce Ci teraz mieszać w głowie, także pominiemy czemu tak się dzieje. 62 powinno wyglądać tak: `intervalId = setInterval(movingHeading, clicks);`  [bez `var`]